src/tests/math/test_linspace.f90

```diff
@@ -0,0 +1,396 @@
+program test_linspace
+    use stdlib_error, only: check
+    use stdlib_kinds, only: sp, dp, int16, int8
+    use stdlib_math, only: linspace, DEFAULT_LINSPACE_LENGTH
+
+    implicit none
+
+    integer :: iunit
+    logical :: warn = .false.
+    real(dp), parameter :: TOLERANCE = 0.00001 ! Percentage of the range for which the actual gap must not exceed
```

Comment:
```suggestion
    real(sp), parameter :: TOLERANCE_SP = 1000 * epsilon(1._sp)  ! Percentage of the range for which the actual gap must not exceed
     real(dp), parameter :: TOLERANCE_dp = 1000 * epsilon(1._dp)  ! Percentage of the range for which the actual gap must not exceed
```

The different values should be appropriately in the rest of the code

Reply:
I will incorporate this change and propagate the appropriate tolerance precision throughout the rest of the test suite.

Reply:
Successfully incorporated 👍